File: Module.Persist.TPM/Model/DTO/PlanPostPromoEffectReport.cs
using Core.Data;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Module.Persist.TPM.Model.DTO {
    public class PlanPostPromoEffectReport : IEntity<Guid>
    {
        public Guid Id { get; set; }
        public string ZREP { get; set; }
        public string DemandCode { get; set; }
        public string PromoNameId { get; set; }
        public int PromoNumber { get; set; }
        public string LocApollo { get; set; }
        public string TypeApollo { get; set; }
        public string ModelApollo { get; set; }
        public DateTimeOffset? WeekStartDate { get; set; }
        public double? PlanPostPromoEffectQty { get; set; }
		public double? PlanUplift { get; set; }
		public DateTimeOffset? StartDate { get; set; }
        public DateTimeOffset? EndDate { get; set; }
		public string Week { get; set; }
		public string Status { get; set; }
		public double? PlanProductBaselineCaseQty { get; set; }
		public double? PlanProductPostPromoEffectLSV { get; set; }
		public double? PlanProductBaselineLSV { get; set; }
		public bool? InOut { get; set; }
        public bool IsOnInvoice { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/planpostpromoeffectreport/PlanPostPromoEffectReport.js
Ext.define('App.model.tpm.planpostpromoeffectreport.PlanPostPromoEffectReport', {
    extend: 'Ext.data.Model',
    idProperty: 'Id',
    breezeEntityType: 'PlanPostPromoEffectReportWeekView',
    fields: [
        { name: 'Id', hidden: true },
        { name: 'ZREP', type: 'string', hidden: false, isDefault: true },
        { name: 'DemandCode', type: 'string', hidden: false, isDefault: true },
        { name: 'PromoNameId', type: 'string', hidden: false, isDefault: true },
        { name: 'LocApollo', type: 'string', hidden: false, isDefault: true },
        { name: 'TypeApollo', type: 'string', hidden: false, isDefault: true },
        { name: 'ModelApollo', type: 'string', hidden: false, isDefault: true },
        { name: 'WeekStartDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'PlanUplift', type: 'number', hidden: false, isDefault: true },
		{ name: 'StartDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
		{ name: 'EndDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
		{ name: 'Status', type: 'string', hidden: false, isDefault: true, defaultFilterConfig: getPlanPostPromoEffectReportDefaultFilterStatus() },
        { name: 'Week', type: 'string', hidden: false, isDefault: true },
        { name: 'PlanPostPromoEffectQtyW1', type: 'number', hidden: false, isDefault: true },
        { name: 'PlanProductBaselineCaseQtyW1', type: 'number', hidden: false, isDefault: true },
        { name: 'PlanProductPostPromoEffectLSVW1', type: 'number', hidden: false, isDefault: true },
        { name: 'PlanProductBaselineLSVW1', type: 'number', hidden: false, isDefault: true },
        { name: 'PlanPostPromoEffectQtyW2', type: 'number', hidden: false, isDefault: true },
        { name: 'PlanProductBaselineCaseQtyW2', type: 'number', hidden: false, isDefault: true },
        { name: 'PlanProductPostPromoEffectLSVW2', type: 'number', hidden: false, isDefault: true },
        { name: 'PlanProductBaselineLSVW2', type: 'number', hidden: false, isDefault: true },

        { name: 'InOut', type: 'bool', hidden: false, isDefault: true },
        { name: 'IsOnInvoice', type: 'bool', hidden: false, isDefault: true },
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'PlanPostPromoEffectReports',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    },
});

function getPlanPostPromoEffectReportDefaultFilterStatus() {
    var statuses = ['Approved', 'Planned'];
    var value = Ext.create('App.extfilter.core.ValueList', statuses);

    var result = {
        value: value,
        operation: 'In'
    };
    return result;
}